src/stream_parser/file_reader.rs

```diff
@@ -452,6 +471,32 @@ impl MaybeRepeatedType {
     }
 }
 
+#[derive(Debug)]
+struct FlagBits {
+    compat_flags: [u8; 8],
+    incompat_flags: [u8; 8],
+    appended_offsets: [u64; 3],
+}
+
+fn parse_flag_bits(message: &model::ULogMessage) -> Result<FlagBits, UlogParseError> {
+    if message.data().len() < 40 {
```

Comment:
Is the <= 40 limit from somewhere? I'd like to throw a comment to related doc for future reference. Or at least name the magic numbers.

Reply:
Yes on the ulog docs: https://docs.px4.io/main/en/dev_log/ulog_file_format.html#b-flag-bits-message
I can extract it to a constant.